### ISSUE TITLE
Insta-crash in file relating to peep rejoin queue.

### ISSUE THREAD

**Issue by wolfreak99** [OpenRCT2/OpenRCT2#3675]
**OS:** Windows 8.1
**Version:** 0.0.5 develop build 9081ddc 
**Commit:** 

Explanation of the issue...
Loading the file causes immediate hang, pausing the debugger in VS pointrs it to an infinite loop in the final "for" loop at the end of "peep_update_ride_sub_state_2_rejoin_queue" in peep.c

```
`   rct_peep* queue_peep;
    for (queue_peep = GET_PEEP(current_last);
        queue_peep->next_in_queue != 0xFFFF;
        queue_peep = GET_PEEP(queue_peep->next_in_queue));

    queue_peep->next_in_queue = peep->sprite_index;
```

`
- [ ] Reproducible in RCT2 (vanilla)? Not sure
- [ ] Multiplayer? It was done in multiplayer (specifically a server ran on the Wolfmod version), but reloading the file in single player on the vanilla openrct2 crashes as well

**Steps to reproduce:**
1. Load the file attached
2. See step 1.

**Dump file**
If you have a dump file, zip it and drag&drop it here.

**Screenshots / Video:**
Drag / drop screenshots here. Use https://vid.me to upload video.

**Save game:**
Change the file extension to .txt or package to a .zip so that it can be drag / dropped here...
[autosave_2016-05-19_20-21-44.zip](https://github.com/OpenRCT2/OpenRCT2/files/273819/autosave_2016-05-19_20-21-44.zip)

**Comment by Chaosmeister** [comment]
I'm unable to reproduce this crash.

**Comment by IntelOrca** [comment]
This was probably fixed by 146982d4b096b42d2435f9d880de49a3734f0f8f
